### ISSUE TITLE
Leaving Brock a rock

### ISSUE THREAD

**Issue by dbbnrl** [i7/counterfeit-monkey#115]
Playing release 7:

I did something the game was clearly not expecting after finding the rock by the t-inserter.  I went back east without restoring him, and encountered Alex's father.  I got the speech that ends in the ultimatum about somebody getting arrested, but I didn't wind up in the cold storage room.  Instead I saw:

Best to start looking for Brock where we know he went: the equipment testing room.

The game has blocked Alex's father from pushing me SE, and I'm still in the hallway.  At this point, I can go back west to the t-inserter, gel Brock, talk with him & play with the machine, and have him transform back to a rock.  Now I can leave that room and exit the Bureau the way I came (by going south, and eventually up to the rotunda).

Now I am outside the Bureau and I have Brock the rock in my inventory!  Since the game will not let me gel him at this point, I've think I've done something it didn't anticipate.

**Comment by angstsmurf** [comment]
I must be missing something, because I can't reproduce this.

[There is code that is meant to prevent leaving the Equipment Testing Room before changing the rock](https://github.com/i7/counterfeit-monkey/blob/master/Counterfeit%20Monkey.materials/Extensions/Counterfeit%20Monkey/Act%20IV%20Among%20Policemen.i7x#L725-L734), and in my testing, it does: trying to leave before gelling the rock gives me "We shouldn't leave here until we have some clue about what happened to Brock when he came in."

All I can think of is that something has seriously broken the table of all known facts. Do you happen to have any save files or transcripts that could help in investigating this?

**Comment by angstsmurf** [comment]
[Here](https://www.dropbox.com/s/n8nfrkr45yb0zqp/Counterfeit%20Monkey%20Test.gblorb?dl=0) is a debug build of the game. After the initial questions, typing TEST BROCKBUG should take you through the game up to the Bureau Basement Secret Section, and also create a transcript of all output text. Depending on your interpreter, you might have to wait a couple of minutes for the script to finish and then press space several times to get back to the input prompt.

If you can't reproduce the bug after this, well, at least it might give us some clues.

**Comment by dbbnrl** [comment]
Looking through the code, here's what I must have done:

- Enter testing room
- Get rock
- Wear monocle
- Examine rock (this sets "Find Brock", Among Policemen line 754)
- Go east, encounter father, finish conversation, get pushed southwest
- Attempt to go SW fails with "Best to start looking..." because in World Model Tweaks line 752, it tests for "Brock-argument", which hasn't happened yet.

I just re-tested this sequence from save and I can duplicate the result.

The fundamental problem, IMHO, is the use of two different test conditions "Find Brock" and "Brock-argument".  If the same condition (either condition!) was used in both places, I suspect the problem could not occur.

**Comment by dbbnrl** [comment]
Sorry, accidental close

**Comment by angstsmurf** [comment]
Right, thanks! I forgot about the monocle. That's an easy fix then. The only question is whether we should block the player from leaving the Equipment Testing Room before gelling the rock, or just allow them to carry it out, bypassing the T-inserter puzzle entirely.

**Comment by dbbnrl** [comment]
Well, I got into the situation because once I confirmed to my satisfaction that the rock was Brock, I reasoned that it might be unsafe to gel him in inside the Bureau.  So, I decided to smuggle him out in my pack, and gel him outside.

The basic question is whether to support this player choice.  My feeling is that the Brock-argument scene is very important narratively (the author comments in the source support this as well).  So, we may not want to allow the player to skip it.  

I think the best fix would be to add a new clause to the sanity check you linked, right after the "player does not know brock-found" clause.  If the Brock-argument hasn't happened, don't let them leave the room -- instead say something about how we'd better talk to Brock first & find out what happened.

**Comment by angstsmurf** [comment]
You're probably right. I won't be home for a couple of days, so I'll probably not have time to work on this until then, but feel free to make a pull request!
